frontend/javascripts/types/api_flow_types.ts

```diff
@@ -976,28 +985,18 @@ export type VoxelyticsWorkflowReport = {
   };
 };
 
-export type VoxelyticsWorkflowListingRun = (
-  | {
-      state: VoxelyticsRunState.RUNNING;
-      beginTime: Date;
-      endTime: null;
-    }
-  | {
-      state:
-        | VoxelyticsRunState.COMPLETE
-        | VoxelyticsRunState.FAILED
-        | VoxelyticsRunState.CANCELLED
-        | VoxelyticsRunState.STALE;
-      beginTime: Date;
-      endTime: Date;
-    }
-) & {
```

Comment:
I understand that I took away some complexity. But the type is only used in `workflow_view.tsx `and `api_flow_types.ts`, and for the workflow view there used to be a `as any as VoxelyticsWorkflowListingRun` cast which made this distinction here a little bit irrelevant if I understand correctly. 
Thus I removed the cast (see below) and widened  the type, so that it can be used without casting. I think that the types of `beginTime` and `endTime` (a few lines below) forbid the `VoxelyticsRunState.PENDING` and `VoxelyticsRunState.SKIPPED` values anyway. 
Maybe there is an even better solution, then I will be happy to learn about it!

Reply:
> and for the workflow view there used to be an as any as Array<RenderRunInfo> cast which made this distinction here a little bit irrelevant if I understand correctly

The cast is still there, though? the old way was a bit more precise and could lead to case where you know it's a "complete" state and therefor endTime must not be null. in general, I would want to keep these distinctions unless they cause a major pain.

Reply:
ah, you meant the cast in `parseRunInfo` in `workflow_list_view.tsx`, right? I removed that too. Now I only have type annotations left, or am I mistaking?
I was trying to keep the types as they were before but I am running into the same problems that probably lead to the cast...